cartographer_ros/cartographer_ros/offline_node_main.cc

```diff
@@ -124,6 +129,16 @@ void Run(const std::vector<string>& bag_filenames) {
     static_tf_broadcaster.sendTransform(urdf_transforms);
   }
 
+  ros::AsyncSpinner async_spinner(kSingleThreaded);
+  async_spinner.start();
+  rosgraph_msgs::Clock clock;
+  auto clock_republish_timer = node.node_handle()->createWallTimer(
```

Comment:
The advantage is that we don't have to wait for trajectory finalization to complete (which can take a while for larger bags) to start republishing the clock. I have a use case where I constantly need the clock published, without dropouts caused by the main thread having to wait for finalization.